clients/js/src/provider.ts

```diff
@@ -43,6 +44,7 @@ export function isLegacyProvider<T extends object>(
 
 export interface SapphireWrapOptions {
   fetcher: KeyFetcher;
```

Comment:
Options object is optional. When provided, TS requires fetcher, but we should be able just to set `enableSapphireSnap` flag. As fetcher always defaults to `new KeyFetcher() ` should it now be `fetcher: KeyFetcher | undefined;` ?